src/main/kotlin/com/example/flashcardbackend/deckgroup/DeckGroupRepository.kt

```diff
@@ -1,55 +1,83 @@
 package com.example.flashcardbackend.deckgroup
 
+import com.example.flashcardbackend.deck.DeckListItem
 import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.dao.DataAccessException
 import org.springframework.jdbc.core.JdbcTemplate
-import org.springframework.jdbc.core.RowMapper
+import org.springframework.jdbc.core.ResultSetExtractor
 import org.springframework.stereotype.Repository
 import org.springframework.transaction.annotation.Transactional
 import java.sql.ResultSet
 import java.sql.SQLException
 
-class DeckGroupRowMapper : RowMapper<DeckGroupListItem> {
-    @Throws(SQLException::class)
-    override fun mapRow(resultSet: ResultSet, i: Int): DeckGroupListItem {
-        return DeckGroupListItem(
-            resultSet.getInt("id"),
-            resultSet.getString("collection_name"),
-            resultSet.getString("description"),
-        )
-    }
-}
-
 @Transactional
 @Repository
 class DeckGroupRepository(@Autowired val jdbcTemplate: JdbcTemplate) {
 
     fun findAll(): List<DeckGroupListItem> {
-        return jdbcTemplate.query("select * from deck_group", DeckGroupRowMapper())
+        return jdbcTemplate.query(
+            "SELECT * FROM deck_group",
+        ) { resultSet, rowNum ->
+            println("Found $rowNum deck groups")
+            DeckGroupListItem(
+                resultSet.getInt("id"),
+                resultSet.getString("group_name"),
+                resultSet.getString("description"),
+            )
+        }
     }
 
     fun insert(deckGroup: DeckGroupCreate): Int {
         return jdbcTemplate.update(
-            "insert into deck_group (group_name, description) values(?,?)",
+            "INSERT INTO deck_group (group_name, description) VALUES (?,?)",
             deckGroup.name,
             deckGroup.description,
         )
     }
 
     fun update(deckGroup: DeckGroupUpdate): Int {
         return jdbcTemplate.update(
-            "insert into deck_group (group_name, description) values(?,?) where id = ?",
+            "UPDATE deck_group SET group_name = ?, description = ? WHERE id = ?",
             deckGroup.name,
             deckGroup.description,
             deckGroup.id,
         )
     }
 
     fun deleteById(id: Int): Int {
-        return jdbcTemplate.update("delete deck_group where id = ?", id)
+        return jdbcTemplate.update("DELETE deck_group WHERE id = ?", id)
     }
 
     fun findById(id: Int): DeckGroup? {
-        // TODO: return deckGroup with list of decks
-        return null
+        val sql = """
+            SELECT deck_group.*, deck.* 
+            FROM deck_group, deck 
+            WHERE deck_group.id = deck.group_id AND deck_group.id = ?
+        """.trimIndent()
+        return jdbcTemplate.query(sql, DeckGroupResultSetExtractor(), id)
+    }
+}
+
```

Comment:
Like I mentioned earlier, the SimpleFlatMapper should help you get rid off this resultset extractor.